src/redisraft.c

```diff
@@ -1008,6 +1008,104 @@ static int cmdRaftRandom(RedisModuleCtx *ctx,
     return REDISMODULE_OK;
 }
 
+#ifdef HAVE_TLS
+/* Callback for passing a keyfile password stored as a char * to OpenSSL,  copied from redis */
+static int tlsPasswordCallback(char *buf, int size, int rwflag, void *u) {
+    UNUSED(rwflag);
+
+    const char *pass = u;
+    size_t pass_len;
+
+    if (!pass) return -1;
+    pass_len = strlen(pass);
+    if (pass_len > (size_t) size) return -1;
+    memcpy(buf, pass, pass_len);
+
+    return (int) pass_len;
+}
+
+SSL_CTX *generateSSLContext(RedisModuleCtx *ctx, RedisRaftCtx *rr) {
+    SSL_CTX *ssl_ctx = SSL_CTX_new(SSLv23_client_method());
+    if (!ssl_ctx) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_CREATE_FAILED");
+//        if (error) *error = REDIS_SSL_CTX_CREATE_FAILED;
+        return NULL;
+    }
+
+    SSL_CTX_set_options(ssl_ctx, SSL_OP_NO_SSLv2 | SSL_OP_NO_SSLv3);
+    SSL_CTX_set_verify(ssl_ctx, SSL_VERIFY_PEER, NULL);
+
+    if ((rr->config->tls_cert != NULL && rr->config->tls_key == NULL) ||
+            (rr->config->tls_key != NULL && rr->config->tls_cert == NULL)) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_CERT_KEY_REQUIRED");
+//        if (error) *error = REDIS_SSL_CTX_CERT_KEY_REQUIRED;
+        goto error;
+    }
+
+    if (!SSL_CTX_load_verify_locations(ssl_ctx, rr->config->tls_ca_cert, NULL)) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_CA_CERT_LOAD_FAILED: tls_ca_ccert = %s", rr->config->tls_ca_cert);
+//        if (error) *error = REDIS_SSL_CTX_CA_CERT_LOAD_FAILED;
+        goto error;
+    }
+
+    if (!SSL_CTX_use_certificate_chain_file(ssl_ctx, rr->config->tls_cert)) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_CLIENT_CERT_LOAD_FAILED: tls_cert = %s", rr->config->tls_cert);
+//        if (error) *error = REDIS_SSL_CTX_CLIENT_CERT_LOAD_FAILED;
+        goto error;
+    }
+    if (!SSL_CTX_use_PrivateKey_file(ssl_ctx, rr->config->tls_key, SSL_FILETYPE_PEM)) {
+        RedisModule_Log(ctx, REDIS_WARNING, "REDIS_SSL_CTX_PRIVATE_KEY_LOAD_FAILED");
+//        if (error) *error = REDIS_SSL_CTX_PRIVATE_KEY_LOAD_FAILED;
+        goto error;
+    }
+
+    if (*rr->config->tls_key_pass != 0) {
```

Comment:
```suggestion
    if (*rr->config->tls_key_pass != '\0') {
```

Let's make it obvious and compare with char.  Btw, `tls_key_pass` can be NULL right? If so, this line would crash.

Reply:
fixed both issues.